salt/_states/caasp_etcd.py

```diff
@@ -64,15 +66,23 @@ def member_add(name, **kwargs):
 def member_remove(name, **kwargs):
     '''
     Remove this node from the etcd cluster
+
+    Arguments:
+
+    * `nodename`: (optional) the nodename for the member we
+                  want to remove. if no name is provided (or empty),
+                  the local node will be used.
     '''
-    this_member_id = __salt__['caasp_etcd.get_member_id']()
-    if not this_member_id:
+    nodename = kwargs.pop('nodename', '')
```

Comment:
Same here, I would just add a `nodename` arg set to be `None` by default instead of relying on `kwargs`. But again, this is personal taste.

Reply:
I agree.